src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNodeTests.cs

```diff
@@ -342,6 +343,132 @@ public void TestGetAllDirectivesUsingDescendantNodes()
+            TestContainsHelper1("#if false\r\n#elif false", SyntaxKind.IfDirectiveTrivia, SyntaxKind.ElifDirectiveTrivia);
+            TestContainsHelper1("#elif true", SyntaxKind.BadDirectiveTrivia);
+            TestContainsHelper1("#if true\r\n#else", SyntaxKind.IfDirectiveTrivia, SyntaxKind.ElseDirectiveTrivia);
+            TestContainsHelper1("#else", SyntaxKind.BadDirectiveTrivia);
+            TestContainsHelper1("#if true\r\n#endif", SyntaxKind.IfDirectiveTrivia, SyntaxKind.EndIfDirectiveTrivia);
+            TestContainsHelper1("#endif", SyntaxKind.BadDirectiveTrivia);
+            TestContainsHelper1("#region\r\n#endregion", SyntaxKind.RegionDirectiveTrivia, SyntaxKind.EndRegionDirectiveTrivia);
+            TestContainsHelper1("#endregion", SyntaxKind.BadDirectiveTrivia);
+            TestContainsHelper1("#error", SyntaxKind.ErrorDirectiveTrivia);
+            TestContainsHelper1("#if true", SyntaxKind.IfDirectiveTrivia);
+            TestContainsHelper1("#nullable enable", SyntaxKind.NullableDirectiveTrivia);
+            TestContainsHelper1("#region enable", SyntaxKind.RegionDirectiveTrivia);
+            TestContainsHelper1("#undef x", SyntaxKind.UndefDirectiveTrivia);
+            TestContainsHelper1("#warning", SyntaxKind.WarningDirectiveTrivia);
+
+            // !# is special and is only recognized at start of a script file and nowhere else.
+            TestContainsHelper2(new[] { SyntaxKind.ShebangDirectiveTrivia }, SyntaxFactory.ParseCompilationUnit("#!command", options: TestOptions.Script));
+            TestContainsHelper2(new[] { SyntaxKind.BadDirectiveTrivia }, SyntaxFactory.ParseCompilationUnit(" #!command", options: TestOptions.Script));
+            TestContainsHelper2(new[] { SyntaxKind.BadDirectiveTrivia }, SyntaxFactory.ParseCompilationUnit("#!command", options: TestOptions.Regular));
+
+            return;
+
+            void TestContainsHelper1(string directive, params SyntaxKind[] directiveKinds)
+            {
+                Assert.True(directiveKinds.Length > 0);
+
+                // directive on its own.
+                TestContainsHelper2(directiveKinds, SyntaxFactory.ParseCompilationUnit(directive));
+
+                // Two of the same directive back to back.
+                TestContainsHelper2(directiveKinds, SyntaxFactory.ParseCompilationUnit($$"""
+                    {{directive}}
+                    {{directive}}
+                    """));
+
+                // Two of the same directive back to back with additional trivia
+                TestContainsHelper2(directiveKinds, SyntaxFactory.ParseCompilationUnit($$"""
+                       {{directive}}
+                       {{directive}}
+                    """));
+
+                // Directive inside a namespace
+                TestContainsHelper2(directiveKinds, SyntaxFactory.ParseCompilationUnit($$"""
+                    namespace N
+                    {
+{{directive}}
```

Comment:
trying to page my brain back in.  good argument that this should not be allowed.  but i think the prefix-checks only occur for the text-node when present, no rules about the content-holes :-/

Reply:
ok.  paged back in.  i think it's because we effectively said that with holes, any restrictions is pointless.  e.g. you can totally write:

```c#
        var x = $"""
            foo{
 1 + 2
            }bar
            """;
```

We don't want to enforce whitespace rules within the hole (Since the hole only creates a value and is unaffected by the prefix stripping).  In other words  the above would generate the same as:

```c#
        var x = $"""
            foo{
                  1 + 2
            }bar
            """;
```

Because teh stripping has no effect on content, there's no point in having rules on it for the hole itself as well.  so if you want to write:

```c#
        var x = $"""
            foo
{1 + 2}
            bar
            """;
```

that's fine.

Reply:
I can't say I agree with this reasoning. When I look at this code, I actually do not know what happens. With the example of
```cs
            $"""
            foo{
                  1 + 2
            }bar
            """;
```
I know that the `3` is going to be inserted after the `foo`, because that's where the interpolation starts. That isn't the case with
```cs
            $"""
            foo
{1 + 2}
            bar
            """;
```
The 3 can't be inserted where the interpolation starts, because that would produce an error about missing whitespace. So where does it go?

Regardless, please indent this code so it's unambiguous what is happening here.

Reply:
Yes.  I see your point.  Looks like an unfortunate missed case with the raw-strings work.  I would not be opposed to bringing this to see if we should disallow.  but my caring level is near 0 :)